src/swsssdk/configdb.py

```diff
@@ -2,29 +2,83 @@
     config_db = ConfigDBConnector()
     config_db.connect()
-    config_db.set_table('bgp_neighbor', '10.0.0.1', {
+    config_db.set_entry('BGP_NEIGHBOR', '10.0.0.1', {
         'admin_status': state
         })
 
-TODO:
-    Will add API on keyspace subscription for incremental configuration support
-    in the incoming versions.
+    # Daemon to watch config change in certain table:
+    config_db = ConfigDBConnector()
+    handler = lambda table, key, data: print (key, data)
+    config_db.add_handler('BGP_NEIGHBOR', handler)
+    config_db.connect()
+    config_db.listen()
+
 """
 
 from .dbconnector import SonicV2Connector
 
 class ConfigDBConnector(SonicV2Connector):
 
     def __init__(self):
-        """Connect to Redis through TCP, which does not requires root.
-        """
+        # Connect to Redis through TCP, which does not requires root.
         super(ConfigDBConnector, self).__init__(host='127.0.0.1')
+        self.handlers = {}
 
     def connect(self):
         SonicV2Connector.connect(self, self.CONFIG_DB, False)
 
+    def add_handler(self, table, handler):
+        """Add a handler to handle config change in certain table.
+        Note that a single handler can be registered to different tables by 
+        calling this fuction multiple times.
+        Args:
+            table: Table name.
+            handler: a handler function that has signature of handler(table_name, key, data)
+        """
+        if not self.handlers.has_key(table):
+            self.handlers[table] = set()
+        self.handlers[table].add(handler)
+
+    def remove_handler(self, table, handler):
+        """Remove a registered handler from a certain table.
+        Args:
+            table: Table name.
+            handler: registered handler.
+        Raises:
+            ValueError: when the handler provided is not registered to the table.
+        """
+        if not self.handlers.has_key(table):
+            raise ValueError("Handler is not binding to this table.")
+        try:
+            self.handlers[table].remove(handler)
+        except:
+            raise ValueError("Handler is not binding to this table.")
+        if len(self.handlers[table]) == 0:
+            self.handlers.pop(table, None)
+
+    def __fire(self, table, key, data):
+        if self.handlers.has_key(table):
+            for handlers in self.handlers[table]:
+                handlers(table, key, data)
+
+    def listen(self):
+        """Start listen Redis keyspace events and will trigger corresponding handlers when content of a table changes.
+        """
+        self.pubsub = self.redis_clients[self.CONFIG_DB].pubsub()
+        self.pubsub.psubscribe("__keyspace@{}__:*".format(self.db_map[self.CONFIG_DB]['db']))
+        for item in self.pubsub.listen():
+            if item['type'] == 'pmessage':
+                _hash = item['channel'].split(':', 1)[1]
+                table = _hash.split(':', 1)[0]
+                key = _hash.split(':', 1)[1]
```

Comment:
in case someone just set a key in the config_db, here you won't get the right message, your listener will crash. It is better to ignore those messages and not to crash.